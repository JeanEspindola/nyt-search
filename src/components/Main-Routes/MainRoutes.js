/*
* MainRoutes Presentation Component
*
* It provides the switch to display the configured routes components.
*
* */
import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Home from '../Home/Home';
import ArticleDetail from '../Article-Detail/ArticleDetail';
import './MainRoutes.css';

const MainRoutes = () => (
  <main className="Main">
    <Switch>
      <Route exact path="/" component={Home} />
      <Route path="/detail" component={ArticleDetail} />
    </Switch>
  </main>
);

export default MainRoutes;
