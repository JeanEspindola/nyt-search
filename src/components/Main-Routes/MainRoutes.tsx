import { Switch, Route } from 'react-router-dom'
import Home from '../../components/Home/Home'
import ArticleDetail from '../Article-Detail/ArticleDetail'
import styles from './MainRoutes.module.scss'

const MainRoutes = () => (
  <main className={styles.container}>
    <div className={styles.contents}>
      <Switch>
        <Route exact path="/" component={Home} />
        <Route path="/detail" component={ArticleDetail} />
      </Switch>
    </div>
  </main>
)

export default MainRoutes
