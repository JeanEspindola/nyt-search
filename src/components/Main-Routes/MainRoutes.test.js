import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import MainRoutes from './MainRoutes';

describe('MainRoutes', () => {
  let Component;

  beforeEach(() => {
    Component = shallow(<MainRoutes />);
  });

  it('renders MainRoutes without crashing', () => {
    expect(Component.length).toBeTruthy();
    expect(shallowToJson(Component)).toMatchSnapshot();
  });
});
