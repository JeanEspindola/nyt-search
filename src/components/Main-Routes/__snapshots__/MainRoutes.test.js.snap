// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainRoutes renders MainRoutes without crashing 1`] = `
<main
  className="Main"
>
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={[Function]}
      path="/detail"
    />
  </Switch>
</main>
`;
